syntax = "proto3";
import "public.proto";
package fixbug;

option cc_generic_services = true;



message LoginRequest
{
    int32 username = 1;
    bytes pwd = 2;
}

message LoginResponse
{
    ResultCode result = 1;
    bool sucess = 2;
    bytes name = 3;
}

message RegisterRequest
{
    uint32 id = 1;
    bytes name = 2;
    bytes pwd = 3;
}

message RegisterResponse
{
    ResultCode result = 1;
    bool sucess = 2;
}
message offlineMsgRequest{
    int32 userid = 1;
}
message offlineMsgResponse{
    repeated bytes msg = 1;
}
service UserServiceRpc
{
    rpc Login(LoginRequest) returns(LoginResponse);
    rpc Register(RegisterRequest) returns(RegisterResponse);
    rpc Getoffline(offlineMsgRequest) returns(offlineMsgResponse);
}
